Formula/c/codecov-cli.rb

```diff
@@ -0,0 +1,158 @@
+    sha256 "e60305c5e5d314f5389259b7f22aaa33d8f7dee49763119234af3755c55b9101"
+  end
+
+  resource "test-results-parser" do
+    url "https://files.pythonhosted.org/packages/fa/1f/35f5b23beb148c36285bfe9722adeaef976814670ebf9bd8129356a283c8/test_results_parser-0.1.0.tar.gz"
+    sha256 "0034281a4b406d7f072fc5ac1f5e44660e3c23bc92f2e7284862ee097f9626ee"
+  end
+
+  resource "tree-sitter" do
+    url "https://files.pythonhosted.org/packages/0d/52/1284e9ed195b161261ac09bfd9785027e2734fc77360a889f6464a8e8ce8/tree_sitter-0.20.2.tar.gz"
+    sha256 "0a6c06abaa55de174241a476b536173bba28241d2ea85d198d33aa8bf009f028"
+  end
+
+  resource "tree-sitter-javascript" do
+    url "https://github.com/tree-sitter/tree-sitter-javascript/archive/refs/tags/v0.20.2.tar.gz"
+    sha256 "5b9d8ae376053d7e6d2953aeb48edf7151d57d76b9a37c9c4590abec8195bafe"
+  end
+
+  resource "tree-sitter-python" do
+    url "https://github.com/tree-sitter/tree-sitter-python/archive/refs/tags/v0.20.2.tar.gz"
+    sha256 "fc8a893f6e08c7236054e43d0acc407ef757a9f47bbb9d910dbc56055d6b9ae6"
+  end
+
+  resource "urllib3" do
+    url "https://files.pythonhosted.org/packages/8b/00/db794bb94bf09cadb4ecd031c4295dd4e3536db4da958e20331d95f1edb7/urllib3-2.0.6.tar.gz"
+    sha256 "b19e1a85d206b56d7df1d5e683df4a7725252a964e3993648dd0fb5a1c157564"
+  end
+
+  resource "PyYAML" do
+    url "https://files.pythonhosted.org/packages/cd/e5/af35f7ea75cf72f2cd079c95ee16797de7cd71f29ea7c68ae5ce7be1eda0/PyYAML-6.0.1.tar.gz"
+    sha256 "bfdf460b1736c775f2ba9f6a92bca30bc2095067b8a9d77876d1fad6cc3b4a43"
+  end
+
+  def install
+    inreplace "setup.py", "codecovcli = ", "codecov = "
+
+    # Workaround for build failures due to authentication issues when pulling git submodules
+    resource("tree-sitter-javascript").stage (buildpath/"languages/treesitterjavascript")
+    resource("tree-sitter-python").stage (buildpath/"languages/treesitterpython")
+
+    venv = virtualenv_create(libexec)
+    %w[anyio certifi charset-normalizer click h11 httpcore httpx idna ijson regex requests responses
+       rfc3986 sniffio test-results-parser tree-sitter urllib3 PyYAML].each do |r|
+         venv.pip_install resource(r)
+       end
+    venv.pip_install_and_link buildpath
+  end
+
+  test do
+    assert_equal "codecovcli, version #{version}\n", shell_output("#{bin}/codecov --version")
+
+    # Unfortunately `shell_output` doesn't capture standard error
+    output = pipe_stderr_output(["#{bin}/codecov create-report",
+                                 "--commit-sha #{SecureRandom.hex(40)}",
+                                 "--slug test/repo --token #{SecureRandom.hex(10)}"].join(" "))
+
+    ["No config file could be found. Ignoring config.",
+     "Process Report creating complete",
+     "Report creating failed: {\"detail\":\"Not valid tokenless upload\"}"].each do |l|
+       assert_match l, output
+     end
+  end
+
+  private
+
+  def pipe_stderr_output(cmd, result = nil)
```

Comment:
You can just use `shell_output` and redirect the stderr stream with the bash argument

Reply:
I assume that you mean something like the following:
```diff
diff --git a/Formula/c/codecov-cli.rb b/Formula/c/codecov-cli.rb
index 0b4e2a1b0e5..aa3fffee580 100644
--- a/Formula/c/codecov-cli.rb
+++ b/Formula/c/codecov-cli.rb
@@ -130,9 +130,11 @@ class CodecovCli < Formula
     assert_equal "codecovcli, version #{version}\n", shell_output("#{bin}/codecov --version")
 
     # Unfortunately `shell_output` doesn't capture standard error
-    output = pipe_stderr_output(["#{bin}/codecov create-report",
-                                 "--commit-sha #{SecureRandom.hex(40)}",
-                                 "--slug test/repo --token #{SecureRandom.hex(10)}"].join(" "))
+    output = shell_output(["bash -c '#{bin}/codecov create-report",
+                           "--commit-sha #{SecureRandom.hex(40)}",
+                           "--slug test/repo --token #{SecureRandom.hex(10)}",
+                           "'",
+                           "2>&1"].join(" "))
 
     ["No config file could be found. Ignoring config.",
      "Process Report creating complete",
@@ -140,19 +142,4 @@ class CodecovCli < Formula
        assert_match l, output
      end
   end
-
-  private
-
-  def pipe_stderr_output(cmd, result = nil)
-    ohai cmd
-    output = IO.popen(cmd, "r+", err: [:child, :out]) do |pipe|
-      pipe.close_write
-      pipe.read
-    end
-    assert_equal result, $CHILD_STATUS.exitstatus unless result.nil?
-    output
-  rescue Minitest::Assertion
-    puts output if verbose?
-    raise
-  end
 end
```

Reply:
Yeah, except you don't have to call bash. It's what `shell_output` always uses

Reply:
Aha, thanks for clarifying that part.

Reply:
📝 `shell_output` uses the Kernel.` method to run `cmd` so that will use whatever the default shell is configured as.

Reply:
I've made those changes, so please would review them when you have time.